src/directives/markers.js

```diff
@@ -233,6 +234,7 @@ angular.module('ui-leaflet').directive('markers',
 
                 getLayers().then(function(layers) {
                     var _clean = function(models, oldModels){
+                        resetMarkerGroups();
```

Comment:
This does resolve the problem, but does it cause all the markers to flash / rebuild?